providers/tests/microsoft/azure/triggers/test_powerbi.py

```diff
@@ -39,19 +39,22 @@
 
 
 @pytest.fixture
-def powerbi_trigger():
-    trigger = PowerBITrigger(
-        conn_id=POWERBI_CONN_ID,
-        proxies=None,
-        api_version=API_VERSION,
-        dataset_id=DATASET_ID,
-        group_id=GROUP_ID,
-        check_interval=CHECK_INTERVAL,
-        wait_for_termination=True,
-        timeout=TIMEOUT,
-    )
+def powerbi_trigger() -> PowerBITrigger:
```

Comment:
I would rewrite the fixture as this without nested method:

```
@pytest.fixture
def powerbi_trigger(timeout=TIMEOUT, check_interval=CHECK_INTERVAL) -> PowerBITrigger:
    """Fixture for creating a PowerBITrigger with customizable timeout and check interval."""
    return PowerBITrigger(
        conn_id=POWERBI_CONN_ID,
        proxies=None,
        api_version=API_VERSION,
        dataset_id=DATASET_ID,
        group_id=GROUP_ID,
        check_interval=check_interval,
        wait_for_termination=True,
        timeout=timeout,
    )
```

You can then parametrize the fixture in the tests by using the @pytest.mark.parametrize if needed. 

That way you can again do:

`task = asyncio.create_task(powerbi_trigger.run().__anext__())`

instead of:

`task = asyncio.create_task(powerbi_trigger().run().__anext__())`

Reply:
My above suggestion will also fix the mypy issue @potiuk mentioned

Reply:
I've applied the changes above

Reply:
@jscheffl I think we are good now regarding the timeout, this one can be merged

Reply:
Thanks for the rework. Code looks good but... I was executing it locally and at least on my machine I still see runtimes at 30 seconds for two tasks:

```
============================ slowest 100 durations =============================
30.04s call     providers/tests/microsoft/azure/triggers/test_powerbi.py::TestPowerBITrigger::test_powerbi_trigger_run_timeout[10]
30.03s call     providers/tests/microsoft/azure/triggers/test_powerbi.py::TestPowerBITrigger::test_powerbi_trigger_run_timeout[0]
18.57s setup    providers/tests/microsoft/azure/fs/test_adls.py::test_get_fs[mocked_connection0-expected_options0]
5.50s call     providers/tests/microsoft/azure/triggers/test_wasb.py::TestWasbBlobSensorTrigger::test_waiting_for_blob
```

Anything that I need to setup locally to remove the 30s bottleneck?

I also still see 30s runtime in CI results:
https://github.com/apache/airflow/actions/runs/11744851979/job/32721562595?pr=43494#step:7:845

Reply:
I’ll look into it, that’s not normal

Reply:
@dabla , I looked into it, the problem is the parameters aren't getting passed to pytest fixture.

Reply:
I reduced the timeout to just 5 secs only and it seems to work fine. 
<img width="1004" alt="Screenshot 2024-11-08 at 11 09 09 PM" src="https://github.com/user-attachments/assets/0d8c1428-42a3-4bd2-87f1-28c5b705dc8a">

Reply:
This is weird, I’ve ran the test locally before pushing and it was instantanious.  Will check why parameter isn’t taken into acount.

Reply:
Really weired... but I can confirm, the performance is good now - thatnks for the efforts in optimizations!